type mismatch in Loader.cpp causes warning

1>Loader.cpp
1>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.13.26128\include\utility(235): warning C4267: 'initializing': conversion from 'size_t' to 'long', possible loss of data
1>c:\users\cliff\desktop\test_island_world\utilities\loader.cpp(585): note: see reference to function template instantiation 'std::pair<long,long>::pair<unsigned __int64,unsigned __int64,0>(std::pair<unsigned __int64,unsigned __int64> &&) noexcept' being compiled
1>c:\users\cliff\desktop\test_island_world\utilities\loader.cpp(585): note: see reference to function template instantiation 'std::pair<long,long>::pair<unsigned __int64,unsigned __int64,0>(std::pair<unsigned __int64,unsigned __int64> &&) noexcept' being compiled

--- a/Global.cpp
+++ b/Global.cpp
@@ -31,10 +31,10 @@ std::shared_ptr<ParameterLink<std::string>> Global::modePL =
         "mode to run MABE in [run,visualize,analyze]");
 
 std::shared_ptr<ParameterLink<int>> Global::maxLineLengthPL =
-    Parameters::register_parameter("PARAMETER_FILES-maxLineLength", 160,
+    Parameters::register_parameter("PARAMETER_FILES-maxLineLength", 80,
                                    "maximum line length");
 std::shared_ptr<ParameterLink<int>> Global::commentIndentPL =
-    Parameters::register_parameter("PARAMETER_FILES-commentIndent", 45,
+    Parameters::register_parameter("PARAMETER_FILES-commentIndent", 30,
                                    "minimum space before comments");
 
 std::shared_ptr<ParameterLink<std::string>> Global::outputDirectoryPL =

--- a/Global.cpp
+++ b/Global.cpp
@@ -31,10 +31,10 @@ std::shared_ptr<ParameterLink<std::string>> Global::modePL =
         "mode to run MABE in [run,visualize,analyze]");
 
 std::shared_ptr<ParameterLink<int>> Global::maxLineLengthPL =
-    Parameters::register_parameter("PARAMETER_FILES-maxLineLength", 80,
+    Parameters::register_parameter("PARAMETER_FILES-maxLineLength", 160,
                                    "maximum line length");
 std::shared_ptr<ParameterLink<int>> Global::commentIndentPL =
-    Parameters::register_parameter("PARAMETER_FILES-commentIndent", 30,
+    Parameters::register_parameter("PARAMETER_FILES-commentIndent", 45,
                                    "minimum space before comments");
 
 std::shared_ptr<ParameterLink<std::string>> Global::outputDirectoryPL =

--- a/Utilities/Loader.cpp
+++ b/Utilities/Loader.cpp
@@ -542,7 +542,7 @@ std::pair<long, long> Loader::generatePopulation(const std::string file_name) {
 
   auto org_file_data = getAttributeMap(file_name);
   auto file_contents_pair =
-      std::make_pair(all_organisms.size(), org_file_data.size());
+      std::make_pair(long(all_organisms.size()), long(org_file_data.size()));
 
   static const std::regex valid_org_name(R"((.*)_organisms(_\d+)?.csv$)");
   std::smatch match_org;

--- a/Utilities/Parameters.cpp
+++ b/Utilities/Parameters.cpp
@@ -505,6 +505,9 @@ void Parameters::printParameterWithWraparound(std::stringstream &file,
   auto comment =
       entire_parameter.substr(pos_of_comment + 3); // + 3 must be cleaned
 
+  std::regex new_line(R"(\n)");
+  comment = std::regex_replace(comment,new_line,"\n ");
+
   // add as much of the comment as possible to the line
   static const std::regex as_much_of_comment(
       R"(.{1,)" + std::to_string(max_line_length - line.length() - 2) +

--- a/Utilities/Parameters.cpp
+++ b/Utilities/Parameters.cpp
@@ -507,22 +507,22 @@ void Parameters::printParameterWithWraparound(std::stringstream &file,
 
   // add as much of the comment as possible to the line
   static const std::regex as_much_of_comment(
-      R"(.{1,)" + std::to_string(max_line_length - line.length() - 3) +
+      R"(.{1,)" + std::to_string(max_line_length - line.length() - 2) +
       R"(}[^\s]*)");
   std::smatch a_m_c;
   std::regex_search(comment, a_m_c, as_much_of_comment);
-
-  line += a_m_c.str();
+  auto first_comment = a_m_c.str();
+  line += "#" + first_comment.substr(2);
   file << line << '\n';
 
   comment = a_m_c.suffix();
   // write rest of the comments right-aligned with slight padding
   static const std::regex aligned_comments(
-      R"(.{1,)" + std::to_string(max_line_length - comment_indent - 5) +
+      R"(.{1,)" + std::to_string(max_line_length - comment_indent - 3) +
       R"(}[^\s]*)");
   for (auto &m : forEachRegexMatch(comment, aligned_comments)) {
     auto comment_piece = m.str();
-    file << sub_line << "  # " << comment_piece
+    file << sub_line << "# " << comment_piece
          << (comment_piece.back() == '\n' ? "" : "\n");
   }
 } // end  Parameters::printParameterWithWraparound

